x/nexus/keeper/general_message.go

```diff
@@ -26,12 +25,11 @@ func getSentMessageKey(destinationChain exported.ChainName, id string) key.Key {
 }
 
 // GenerateMessageID generates a unique general message ID
-func (k Keeper) GenerateMessageID(ctx sdk.Context) string {
+func (k Keeper) GenerateMessageID(ctx sdk.Context, txHash []byte) string {
```

Comment:
I don't see anywhere using this function but pass `txHash` in that doesn't equal to `sha256.Sum256(ctx.TxBytes())`? If so, adding this extra param looks weird to me.

Reply:
We need the hash to construct the `Message` struct anyways, so I thought it made sense to just pass it in here as well, instead of computing the hash twice.

Reply:
I don't feel strongly though, because it's pretty cheap to compute the hash, and so we could just compute the hash twice in order to keep the code clean.

Reply:
I don't think that cost is worth it to "pollute" the function signature.